### PR TITLE
Shader dev 

### PR DESCRIPTION
**Pull request by LegendWasTaken**
Compute shaders are now at a stage where a decent amount of work has been done to them, allowing for scenes and camera movement.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -7,7 +7,7 @@ find_package(Vulkan REQUIRED)
 add_subdirectory(external)
 
 add_executable(rvpt src/rvpt/main.cpp src/rvpt/rvpt.cpp src/rvpt/window.cpp src/rvpt/vk_util.cpp
-    assets/shaders/fullscreen_tri.vert assets/shaders/tex_sample.frag)
+    assets/shaders/fullscreen_tri.vert assets/shaders/tex_sample.frag src/rvpt/camera.cpp src/rvpt/camera.h src/rvpt/timer.cpp src/rvpt/timer.h)
```
Comment (cdgiessen) [inline CMakeLists.txt:10]: No need to add the .h files.

File: assets/shaders/structs.glsl [hunk 2]
```diff
@@ -10,10 +10,13 @@ struct Ray
...
-    vec3 vertical;
-};
+    bool hit;
+    float distance;
+    float reflectiveness;
+    vec3 emission;
+    vec3 albedo;
+    vec3 normal;
+    vec3 intersection;
+};
```
Comment (cdgiessen) [inline assets/shaders/structs.glsl:22]: Should have a blank new line at the end (so the red cancel mark goes away)

File: src/rvpt/camera.h [hunk 3]
```diff
@@ -0,0 +1,26 @@
+//
+// Created by legend on 5/26/20.
+//
+
+#pragma once
+
+#include <glm/glm.hpp>
+#include <vector>
+
+class camera
```
Comment (cdgiessen) [inline src/rvpt/camera.h:10]: `camera` should be capitalized.

File: src/rvpt/main.cpp [hunk 4]
```diff
@@ -29,6 +56,8 @@ int main()
             window.set_close();
         if(window.is_key_down(Window::KeyCode::KEY_R))
             rvpt.reload_shaders();
+        update_camera(window, rvpt);
+        std::cout << rvpt.time.average_frame_time() << '\n';
```
Comment (cdgiessen) [inline src/rvpt/main.cpp:60]: Don't spam the console with frame data by default. 
Though, this shouldn't be needed anyways once I get imgui running...

File: src/rvpt/rvpt.cpp [hunk 5]
```diff
@@ -6,9 +6,12 @@
 #include <fstream>
 
 #include <glm/glm.hpp>
+#include <glm/glm/ext.hpp>
 #include <nlohmann/json.hpp>
 
-RVPT::RVPT(Window& window) : window_ref(window)
+#include <random>
```
Comment (cdgiessen) [inline src/rvpt/rvpt.cpp:12]: Prefer to put standard library headers above external libraries.
My order is standard C++ libraries, standard C libraries, external libraries, header files not in the local folder, local header files. (basically, the order of 'C++, C, external libraries, internal libraries but not in the same 'module', then in-'module' headers.

File: src/rvpt/timer.cpp [hunk 6]
```diff
@@ -0,0 +1,45 @@
+//
+// Created by legend on 5/27/20.
+//
+
+#include <algorithm>
+#include "timer.h"
```
Comment (cdgiessen) [inline src/rvpt/timer.cpp:6]: timer.h should be included first. (for .cpp files, the header include order is the same as header files, just with the .cpp's header files first.

**Comment by LegendWasTaken** [comment]
Resolved comments.

**Pull request closed without merging**